source/agora/consensus/Fee.d

```diff
@@ -500,8 +540,12 @@ public class FeeManager
         out Amount rate) nothrow @safe
     {
         try
+        {
             // At this point, we get a fee, not a rate, but we divide it below.
-            rate = tx.getFee(peekUTXO);
+            auto freeze_fee = this.getFreezeFee(tx);
+            ensure(freeze_fee.isValid(), "Can't calculate freeze fee");
+            rate = tx.getFee(peekUTXO) - freeze_fee;
```

Comment:
Freeze fees are excluded from rate calculations or they highly skew the minimum fee for TX pool. acceptance

Reply:
Maybe it is better that freeze `tx`s are always chosen before other `tx`s to enable enrolments.

Reply:
I think I understand now. This is to prevent `tx` rejection due to  https://github.com/bosagora/agora/blob/1f67ab377ac085fb70ebe81ec6ff441421a8d624/source/agora/node/Config.d#L240-L266

Reply:
Exactly:)